sqlglot/optimizer/optimizer.py

```diff
@@ -56,7 +57,12 @@ def optimize(expression, schema=None, db=None, catalog=None, rules=RULES, **kwar
     Returns:
         sqlglot.Expression: optimized expression
     """
-    possible_kwargs = {"db": db, "catalog": catalog, "schema": schema or sqlglot.schema, **kwargs}
+    schema = ensure_schema(schema or sqlglot.schema)
+
+    if isinstance(schema, AbstractMappingSchema):
```

Comment:
why do you check the type here?

Reply:
Because if the user supplies a custom `Schema` that doesn't support the `normalize` method, it'll crash. Might not be the best way to do it though, maybe we could check if the schema object has a `normalize` attribute instead? Not sure.

Reply:
normalize should not have to be called, it should be a part of the init